File: app/src/main/java/com/open/learncode/java/泛型/java/继承泛型类实现泛型接口/SubClass.java
package com.open.learncode.java.泛型.java.继承泛型类实现泛型接口;

public class SubClass<T1, T2> extends ExtendClass<T2> implements IImpl<String> {

    @Override
    public void show(T2 t) {
        super.show(t);
    }

    @Override
    public void print(String key) {
        System.out.println("实现泛型接口 IImpl print：" + key);
    }
}

File: app/src/main/java/com/open/learncode/java/泛型/kotlin/继承泛型类实现泛型接口/SubClass.kt
package com.open.learncode.java.泛型.kotlin.继承泛型类实现泛型接口

class SubClass<T1, T2> : ExtendClass<T2>(), IImpl<String> {

    override fun show(t: T2) {
        super.show(t)
    }

    override fun print(key: String) {
        println("实现泛型接口 IImpl print：$key")
    }
}